src/display/annotation_layer.js

```diff
@@ -718,6 +723,72 @@ var StrikeOutAnnotationElement = (
 })();
 
 /**
+ * @class
+ * @alias FileAttachmentAnnotationElement
+ */
+var FileAttachmentAnnotationElement = (
+    function FileAttachmentAnnotationElementClosure() {
+  function FileAttachmentAnnotationElement(parameters) {
+    AnnotationElement.call(this, parameters, true);
+
+    this.filename = PDFJS.getFilenameFromUrl(parameters.data.file.filename);
+    this.content = parameters.data.file.content;
+  }
+
+  Util.inherit(FileAttachmentAnnotationElement, AnnotationElement, {
+    /**
+     * Render the file attachment annotation's HTML element in the empty
+     * container.
+     *
+     * @public
+     * @memberof FileAttachmentAnnotationElement
+     * @returns {HTMLSectionElement}
+     */
+    render: function FileAttachmentAnnotationElement_render() {
+      this.container.className = 'fileAttachmentAnnotation';
+      this.container.addEventListener('click', this._download.bind(this));
```

Comment:
Unfortunately we can't. I tried that, but in JavaScript when the `dblclick` event fires, the `click` event fires automatically too and there is nothing you can do about that (except introducing ugly hacks that are not cross-browser compatible). To reduce the problem, I now attached the event listener to the trigger div instead of the container. This means that you can now close the popup by clicking on it without triggering the download like before. I don't think we can do any better than this: clicking the icon will still trigger the download and the opening of the pop-up.